packages/expo-asset/src/resolveAssetSource.web.ts

```diff
@@ -7,19 +7,18 @@ declare let nativeExtensions: any;
 
 let _customSourceTransformer;
 let _serverURL: string | undefined | null;
-let _scriptURL: string | undefined | null;
 let _sourceCodeScriptURL: string | undefined | null;
 
 function getSourceCodeScriptURL(): string | undefined | null {
   if (_sourceCodeScriptURL) {
     return _sourceCodeScriptURL;
   }
 
-  let sourceCode = nativeExtensions && nativeExtensions.SourceCode;
+  let sourceCode = typeof nativeExtensions !== 'undefined' ? nativeExtensions.SourceCode : null;
   if (!sourceCode) {
-    sourceCode = NativeModules && NativeModules.SourceCode;
+    sourceCode = NativeModules?.SourceCode;
   }
-  _sourceCodeScriptURL = sourceCode.scriptURL;
+  _sourceCodeScriptURL = sourceCode?.scriptURL;
```

Comment:
```suggestion
  return  (nativeExtensions?.SourceCode ?? NativeModules?.SourceCode)?.scriptURL
```
Would let you get rid of 5 lines